### PR TITLE
Bring back renderRect to Sprite

### PR DESCRIPTION
**Pull request by luanpotter**
# Description

Bring back renderRect to Sprite

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors and are passing (See [Contributor Guide]).
- [x] I updated/added relevant documentation (doc comments with `///`) and updated/added examples in `doc/examples`.
- [x] I have formatted my code with `./scripts/format.sh` and the Flame analyzer (`./scripts/analyze.sh`) does not report any problems.
- [x] I read and followed the [Flutter Style Guide].
- [x] I have added a description of the change under `[next]` in `CHANGELOG.md`.
- [x] I removed the `Draft` status, by clicking on the `Ready for review` button in this PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flame users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in `CHANGELOG.md`).
- [x] No, this is *not* a breaking change.

## Related Issues

*Replace this paragraph with a list of issues related to this PR from the [issue database]. Indicate, which of these issues are resolved or fixed by this PR.*

[issue database]: https://github.com/flame-engine/flame/issues
[Contributor Guide]: https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo

### CODE REVIEW

File: packages/flame/lib/src/sprite.dart [hunk 1]
```diff
@@ -54,11 +62,31 @@ class Sprite {
...
+      overridePaint: overridePaint,
+    );
+  }
+
   /// Renders this sprite onto the canvas.
   ///
   /// * position: x,y coordinates where it will be drawn; default to origin.
-  /// * size: width/height dimensions; it can be bigger or smaller than the original size -- but it defaults to the original texture size.
-  /// * overridePaint: paint to use. You can also change the paint on your Sprite instance. Default is white.
+  /// * size: width/height dimensions; it can be bigger or smaller than the original size -- but it
```
Comment (wolfenrain) [inline packages/flame/lib/src/sprite.dart:86]: I know this isn't really part of your PR, but we should annotate these with dartdocs

**Review by wolfenrain — APPROVED** [review]
LGTM

**Review by spydon — APPROVED** [review]

**Review by erickzanardo — APPROVED** [review]
LGTM

File: packages/flame/lib/src/sprite.dart (hunk 1 above)
Comment (luanpotter) [reply packages/flame/lib/src/sprite.dart:86]: ah good catch! missed that

**Pull request merged**